dali/test/python/operator_1/test_operator_origin_trace.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
How about adding a simple test to check if DALI internals are correctly filtered out? This piece of code:
```py
if (
   len(call_context) > 2
   and call_context[-2].filename == current_filename
   and call_context[-2].name == "__enter__"
   and call_context[-3].filename == current_filename
    and call_context[-3].name == "_generate_graph"
):
```
is a heuristic and I can imagine it might stop working one day when somebody for example adds a helper function between `_generate_graph` and `__enter__`. A test for that would allow us to notice it faster

Reply:
The tests already check against an expected stack trace, captured by hand (or by running the pipeline definition as normal function) so I think this should already be covered.

Reply:
That's right, I thought they only check if the stack trace _contains_ the pattern, but now I see they look for exact match. Sorry!